backend/src/api/v1/auth/routes.py

```diff
@@ -1,40 +1,50 @@
-from typing import Any
-
 from fastapi import APIRouter, HTTPException, status
 
-from src.api.deps import OAuth2Form, TransactionalSession
-from src.api.v1.auth.schemas import TokenPayload, UserCreate
-from src.api.v1.auth.service import (
-    create_user,
-    get_user_by_email,
-    is_email_registered,
-)
+from src.api.deps import Session
+from src.api.v1.auth.deps import PasswordForm
+from src.api.v1.auth.schemas import Token
+from src.api.v1.schemas import Message
+from src.api.v1.users.schemas import UserCreate, UserPasswordReset
+from src.api.v1.users.service import create_user, get_user_by_email, is_email_registered, update_password
+from src.api.v1.verification.service import expire_code_if_valid
 from src.core.security import create_access_token, is_valid_password
 
 router = APIRouter(prefix="/auth", tags=["Authentication"])
 
 
-@router.post("/register", status_code=status.HTTP_201_CREATED, response_model=TokenPayload)
-async def register(user_create_model: UserCreate, session: TransactionalSession) -> Any:
+@router.post("/register", status_code=status.HTTP_201_CREATED)
+async def register(user_create_model: UserCreate, session: Session) -> Token:
     if is_email_registered(session, user_create_model.email):
         raise HTTPException(status.HTTP_409_CONFLICT, "Email already registered")
 
     user = create_user(session, user_create_model)
-    access_token = create_access_token(user.id)
-    return access_token
+    return create_access_token(user.id)
 
 
-@router.post("/login", response_model=TokenPayload)
-async def login(form: OAuth2Form, session: TransactionalSession) -> Any:
-    email = form.username  # The OAuth2 spec requires the exact field name `username`.
+@router.post("/login")
+async def login(form: PasswordForm, session: Session) -> Token:
+    email = form.username  # The OAuth2 spec requires the exact name `username`.
     user = get_user_by_email(session, email)
 
     if not user:
         raise HTTPException(status.HTTP_404_NOT_FOUND, "User not found")
-    if not is_valid_password(form.password, user.password):
-        raise HTTPException(status.HTTP_401_UNAUTHORIZED, "Incorrect password")
     if not user.is_active:
         raise HTTPException(status.HTTP_401_UNAUTHORIZED, "Inactive user")
+    if not is_valid_password(form.password, user.password):
+        raise HTTPException(status.HTTP_401_UNAUTHORIZED, "Incorrect password")
+
+    return create_access_token(user.id)
+
+
+@router.patch("/reset")
```

Comment:
The name is unclear
```suggestion
@router.patch("/reset-password")
```

Reply:
could you please make a github suggestion? i'll accept it